libdnf/module/modulemd/ModuleDefaultsContainer.cpp

```diff
@@ -28,8 +28,11 @@
 namespace libdnf {
 
 ModuleDefaultsContainer::ModuleDefaultsContainer()
+: prioritizer(modulemd_prioritizer_new()) {}
+
+ModuleDefaultsContainer::~ModuleDefaultsContainer()
 {
-    prioritizer = std::shared_ptr<ModulemdPrioritizer>(modulemd_prioritizer_new(), g_object_unref);
+    g_object_unref(prioritizer);
```

Comment:
Why not use a `unique_ptr`? Not a big difference here but I'd still consider it a bit cleaner.

Reply:
I don't see an issue here.

Reply:
I didn't say there was an issue, just that `unique_ptr` would be better.

Reply:
Usually I prefer to use unique_ptr. The first version of this PR contained unique_ptr.
declaration:
`std::unique_ptr<ModulemdPrioritizer, decltype(&g_object_unref)> prioritizer;`
definition in ModuleDefaultsContainer constructor:
`prioritizer(std::unique_ptr<ModulemdPrioritizer, decltype(&g_object_unref)>(modulemd_prioritizer_new(), g_object_unref))`
I was also thinking about 3 other ways how use unique_ptr in the PR. But at the end I decided to made this simple case without unique_ptr.

Reply:
Fair enough, I guess.

We could do something like this, though, and use it for all g_object pointers? (no idea how many we actually have)
```c++
template<class T> using g_unique_ptr = unique_ptr<T, decltype(&g_object_unref)>;

template<class T, class... Args>
g_unique_ptr<T> g_make_unique(Args&&... args) {...}
```

(I don't want to block this PR on this though, just please reference the bug in the relevant commit?)

Reply:
I added (RhBug:1636803) to commit message.